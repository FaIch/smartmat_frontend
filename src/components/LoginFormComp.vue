<template>
  <div class="login-form">
    <div class="title">Logg inn</div>
    <form action="#">
      <div class="input-boxes">
        <div class="input-box">
          <i class="fas fa-envelope"></i>
          <input type="text" placeholder="Email..." required>
        </div>
        <div class="input-box">
          <i class="fas fa-lock"></i>
          <input type="password" placeholder="Passord..." required>
        </div>
        <div class="button input-box">
          <input type="submit" value="Logg inn">
        </div>
        <div class="text sign-up-text">Har du ikke bruker?
          <label @click="emit('switch-view')">Registrer deg</label>
        </div>
      </div>
    </form>
  </div>
</template>

<script setup lang="ts">

const emit = defineEmits(['switch-view'])
</script>

<style scoped>
</style>
