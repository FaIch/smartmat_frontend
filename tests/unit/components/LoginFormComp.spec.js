import { describe, expect, it, beforeEach } from 'vitest'
import { mount } from '@vue/test-utils'
import LoginFormComp from '../../../src/components/LoginFormComp.vue'
import { createPinia } from 'pinia'

const pinia = createPinia()
const mockStore = {
  someGetter: 'mockValue'
}
let wrapper

beforeEach(() => {
  wrapper = mount(LoginFormComp, {
    global: {
      plugins: [pinia],
      mocks: {
        $pinia: pinia,
        $store: mockStore
      }
    }
  })
})

describe('Login form validation', () => {
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can write into email input', async () => {
    const emailInput = wrapper.find('.input-box input[type="email"]')
    await emailInput.setValue('Test')
    expect(emailInput.element.value).toEqual('Test')
  })

  it('should show an error message if email field is empty', async () => {
    const emailInput = wrapper.find('.input-box input[type="email"]')
    const passwordInput = wrapper.find('.input-box input[type="password"]')
    const updateMessage = wrapper.find('.update-message h4')

    await emailInput.setValue('')
    await passwordInput.setValue('testpassword')
    await wrapper.find('.login-form form').trigger('submit')

    expect(updateMessage.text()).toEqual('Vennligst fyll alle felt.')
  })

  it('should show an error message if email field is invalid', async () => {
    const emailInput = wrapper.find('.input-box input[type="email"]')
    const passwordInput = wrapper.find('.input-box input[type="password"]')
    const updateMessage = wrapper.find('.update-message h4')

    await emailInput.setValue('invalidemail')
    await passwordInput.setValue('testpassword')
    await wrapper.find('.login-form form').trigger('submit')

    expect(updateMessage.text()).toEqual('Ugyldig e-post.')
  })

  it('should show an error message if email is too long', async () => {
    const emailInput = wrapper.find('.input-box input[type="email"]')
    const passwordInput = wrapper.find('.input-box input[type="password"]')
    const updateMessage = wrapper.find('.update-message h4')

    await emailInput.setValue('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa@gmail.com')
    await passwordInput.setValue('testpassword')
    await wrapper.find('.login-form form').trigger('submit')

    expect(updateMessage.text()).toEqual('E-post er for lang.')
  })

  it('should show an error message if password is too long', async () => {
    const emailInput = wrapper.find('.input-box input[type="email"]')
    const passwordInput = wrapper.find('.input-box input[type="password"]')
    const updateMessage = wrapper.find('.update-message h4')

    await emailInput.setValue('test@example.com')
    await passwordInput.setValue('a'.repeat(201))
    await wrapper.find('.login-form form').trigger('submit')

    expect(updateMessage.text()).toEqual('Passord er for lang.')
  })

  it('should show an error message if password is too short', async () => {
    const emailInput = wrapper.find('.input-box input[type="email"]')
    const passwordInput = wrapper.find('.input-box input[type="password"]')
    const updateMessage = wrapper.find('.update-message h4')

    await emailInput.setValue('test@example.com')
    await passwordInput.setValue('a'.repeat(3))
    await wrapper.find('.login-form form').trigger('submit')

    expect(updateMessage.text()).toEqual('Passord må være minst 8 karakterer langt.')
  })
})
