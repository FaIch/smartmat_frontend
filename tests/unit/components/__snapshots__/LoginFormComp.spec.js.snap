// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login form validation > renders correctly 1`] = `
"<div data-v-b1606448=\\"\\" class=\\"login-form\\">
  <div data-v-b1606448=\\"\\" class=\\"title\\">Logg inn</div>
  <div data-v-b1606448=\\"\\" class=\\"update-message\\">
    <h4 data-v-b1606448=\\"\\"></h4>
  </div>
  <form data-v-b1606448=\\"\\">
    <div data-v-b1606448=\\"\\" class=\\"input-boxes\\">
      <div data-v-b1606448=\\"\\" class=\\"input-box\\"><i data-v-b1606448=\\"\\" class=\\"fas fa-envelope\\"></i><input data-v-b1606448=\\"\\" data-cy=\\"input-email\\" type=\\"email\\" placeholder=\\"Email...\\" required=\\"\\"></div>
      <div data-v-b1606448=\\"\\" class=\\"input-box\\"><i data-v-b1606448=\\"\\" class=\\"fas fa-lock\\"></i><input data-v-b1606448=\\"\\" data-cy=\\"input-password\\" type=\\"password\\" placeholder=\\"Passord...\\" required=\\"\\"></div>
      <div data-v-b1606448=\\"\\" class=\\"button input-box\\" data-cy=\\"button-login\\"><input data-v-b1606448=\\"\\" type=\\"submit\\" value=\\"Logg inn\\"></div>
      <div data-v-b1606448=\\"\\" class=\\"text sign-up-text\\">Har du ikke bruker? <label data-v-b1606448=\\"\\" data-cy=\\"switch-to-signup\\">Registrer deg</label></div>
    </div>
  </form>
</div>"
`;
